doc/holochain_101/src/planning_a_dapp.md

```diff
@@ -1 +1,69 @@
+Because app data storage is distributed amongst user-participants, one must expect that data encryption and permissions are important for protecting privacy in accordance with the jurisdictions in which the app is operating.
+
+Remember that, as user-participants leave the application, they take their data with them. They also retain copies of other data that they held to support the DHT.  
+
+One must also re-think the dApp's business model such that it does not rely on a central authority's ability to whitelist access to a given resource. 
+
+## Cryptography in Holochain dApps
+Distributed systems rely more heavily on cryptographic patterns and techniques than centralized systems. 
+
+### Hashes
+
+### Signatures 
+
+### Encryption
+
+## Entry types 
+You can define particular entry types as public, private, or encrypted. Encryption is an in-built option for Holochain apps, meaning that you can automatically encrypt entry types by default.
+This is because Holochain's use of DHTs means that simply running the application allows user-participants access to all entries on the network.
+
+### Public entries
+Public entries are published to the app's DHT with an appropriate redundancy factor. 
+
+### Encrypted entries
+Like public entries, encrypted entries are published to the app's DHT with an appropriate redundancy factor.
+
+### Private entries
+Unlike public and encrypted entries, private entries are only on the source chain (device) of the publisher and are not shared out to the DHT. 
+Of course, one could always encrypt a private entry, too. 
+
+## Security - best practices
+A great way to begin offsetting the governance crises now typical of distributed systems (i.e. DAO hack) is to think in terms of protecting and enabling the community of user-participants in addition to cryptography.    
+
+In essence, one must consider how to prevent undesired access to the DHT. If membranes are not properly built in the dApps's DNA, having access to the source code also means having access to the entire network's entries via the DHT. Developers must treat the code, or at least the DNA taken as a whole, as if it's a key to the data. Note, too, that one can easily fork a Holochain dApp without disrupting the its activity, making it possible to retain the benefits of open-source code without some of the risks.
+
+### Membranes
+
+Therefore, security efforts begin with the specification of membranes, lest the code itself become a target. Though holochains rely on the cryptography above to create trust in data's provenance and immutability, trust is a distinctly human affair at the level of creating membranes. Different applications will require different levels of security, and Holochain is uniquely suited to accommodate a high degree of flexibility. DNA could define a closed list of participants, Proof of Service, or social triangulation requirements, for example.  
```

Comment:
Yes.

Not as a request before this can be merge, but mentioning Sybil Attacks here would make sense (https://en.wikipedia.org/wiki/Sybil_attack). Blockchains prevent Sybil Attacks with PoW (as an implied membrane) and we need something similar as well, or otherwise attackers can overrun the DHT and create a constructed world for the victim. Now, the nice thing about our agent-centric approach is that we can do so much more than just PoW (which a Holochain app could implement of course) in order to identify and filter out Sybil nodes (=nodes without history basically).

Reply:
Please don't start the sentence with `Therefore, `